LayoutTests/accessibility/animated-dropdown.html

```diff
@@ -0,0 +1,91 @@
+  </section>
+  <section><button>Foo</button></section>
+</main>
+<footer><a href="#url">apple.com</a></footer>
+<script>
+var output = "This test ensures the accessibility tree is correct after an animated dropdown is opened via button press.\n\n";
+
+if (window.accessibilityController) {
+    window.jsTestIsAsync = true;
+
+    output += "First traversal:\n";
+    output += dumpAXSearchTraversal(accessibilityController.rootElement.childAtIndex(0));
+    var secondTraversal;
+    setTimeout(async function() {
+        await sleep(1000)
+        // Expand the dropdown.
+        accessibilityController.accessibleElementById("button-one").press();
+        // Wait an arbitrary amount of time to allow the accessibility tree to update.
+        output += "\nSecond traversal:\n\n";
+        const expectedText = "This text begins as hidden";
+        await waitFor(() => {
+            secondTraversal = dumpAXSearchTraversal(accessibilityController.rootElement.childAtIndex(0));
+            return secondTraversal && secondTraversal.includes(expectedText);
+        });
+        output += expect(`secondTraversal.includes('${expectedText}')`, "true");
+        if (accessibilityController.platformName !== "ios") {
+            // The button won't have a parent if the implementation bug is present.
+            // iOS doesn't implement AccessibilityUIElement::parentElement (it arguably doesn't make sense to do so for iOS?).
+            output += expect("!!accessibilityController.accessibleElementById('button-one').parentElement()", "true");
+        }
+        output += secondTraversal;
+
+        document.getElementById("main").style.display = "none";
+        debug(output);
+        finishJSTest();
+    }, 0);
+}
+
+(() => {
+    var t = {
+            641: () => {
+                $((function() {})),
+                    function() {}(jQuery), $(document).ready((function() {
+                        $(".dropdown").click((function(a) {
+                            $(this).next().slideToggle()
+                        }))
+                    }))
+            },
+        },
+        e = {};
+
+    function o(a) {
+        var s = e[a];
+        if (void 0 !== s) return s.exports;
+        var l = e[a] = {
+            exports: {}
+        };
+        return t[a](l, l.exports, o), l.exports
+    }
+    o.m = t, a = [], o.O = (t, e, s, l) => {
+        if (!e) {
+            var i = 1 / 0;
+            for (d = 0; d < a.length; d++) {
+                for (var [e, s, l] = a[d], n = !0, c = 0; c < e.length; c++)(!1 & l || i >= l) ? e.splice(c--, 1) : i = l;
+                var r = s();
+            }
+            return t
+        }
+        l = l || 0;
+        for (var d = a.length; d > 0 && a[d - 1][2] > l; d--) a[d] = a[d - 1];
+        a[d] = [e, s, l]
+    }, o.o = (a, t) => Object.prototype.hasOwnProperty.call(a, t), (() => {
+        o.O.j = t => 0 === a[t];
+    })(), o.O(void 0, [], (() => o(641))), o.O();
+    var s = o.O(s);
+})();
```

Comment:
What is this?

Reply:
Minified JS taken from a webpage exhibiting this bug. Originally the corpus was significantly larger, but I spent a ton of time reducing it to this — I couldn't get it any smaller without breaking the behavior under-test.

Reply:
Greatly appreciate that you took the big pain of reducing the test case. I'm still confused about this block of code with lots of 1 letter identifiers and style that don't follow WebKit standards. Can we comment this, or even write it in a more readable way?

Reply:
OK, all of that minified JS is removed in the latest commit.